numpyro/infer/hmc_gibbs.py

```diff
@@ -0,0 +1,108 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from collections import namedtuple
+import copy
+from functools import partial
+
+from jax import device_put, random, value_and_grad
+
+from numpyro.handlers import condition, seed, trace
+from numpyro.infer.mcmc import MCMCKernel
+from numpyro.infer.hmc import HMC, NUTS
+from numpyro.util import ravel_pytree
+
+
+HMCGibbs_State = namedtuple("HMCGibbs_State", "z, hmc_state, rng_key")
```

Comment:
nit: probably `HMCGibbsState` is more similar to `HMCState` but it depends on your preference